#162 Data Annotation

hopetambala opened the pull request:
**Description of proposed changes**
What does this pr accomplish and why should it be integrated?
Implement a note-taking feature that is a panel on the right-hand side that doubles as replacing the tooltip feature (found in d3). It has the ability to create a note with a title, labels, and note content. It then saves that note for a user to have the ability to export the notes and is retrievable on upload dataset.

**Checklist**
_Put an x in the boxes that apply. _

[x ]Lint and unit tests pass locally with my changes
[x ] I have added tests that prove my changes work (if applicable)
[x ]I have added or updated necessary documentation (if appropriate)

**Additional comments**
Is there anything else you want us to know?
Turns off the d3 mouseover

adds a setPosition action to the controls reducer (for a future use)

rashley-iqt commented on src/features/tooltip/Tooltip.js:
```diff
@@ -0,0 +1,206 @@
+  componentDidUpdate = (prevProps) => {
+      if (prevProps.data !== this.props.data) {
+        if(this.props.data.fieldValue){
+          console.log(this.props.data.fieldValue, this.props.data)
```
please remove the console logging line.

rashley-iqt commented on src/features/visualization/d3-viz.js:
```diff
@@ -290,4 +290,4 @@ const findAncestor = (node, hierarchy) => {
 };
 
-export default d3Viz;
+export default d3Viz
```
please reinsert the semi-colon at the end of this line.

rashley-iqt commented on src/features/visualization/d3-viz.js:
```diff
@@ -220,6 +219,7 @@ function d3Viz(rootNode) {
 
     nodeRoot.on("click.select", () => {
       const datum = select(d3Event.target).datum();
+      if(datum.data !== undefined) {props.sendData(datum.data)} //Used to send data back to parent component Visualizations.js
```
please make this if a multi-line to read more consistently.

rashley-iqt commented on src/features/visualization/d3-viz/setup-tooltip.js:
```diff
@@ -1,5 +1,11 @@
-import { map, join, path } from "ramda";
-import { select, event as d3Event } from "d3-selection";
+/*
```
If the intent is to remove the reliance on this tooltip then actually delete this file and remove any references instead of leaving dead/non-executing code in place.

hopetambala commented:
Addressed these issues

- [x] Linting errors 
- [x] Hid the detail panel on the initial load until there is something to show the details of. 
- [x] Save/delete buttons are:
    * centered
   * black and white 
   * consistent with the size/style of the other icons
- [x] Visually offset the notes area from the imported data with CRViz style conventions
- [x] Once a note is entered, a user doesn't have to click off of the node and then back on to see the note. 
- [x] Made visually clear that the title and note areas are text areas are text fields
- [x] Made the notes entry area fit the width of the tooltip through multi-line 
- [x] Remove Tagging feature
